src/components/AttachmentCarousel/index.js

```diff
@@ -0,0 +1,215 @@
+import * as Expensicons from '../Icon/Expensicons';
+import styles from '../../styles/styles';
+import themeColors from '../../styles/themes/default';
+import CarouselActions from './CarouselActions';
+import Button from '../Button';
+import * as ReportActionsUtils from '../../libs/ReportActionsUtils';
+import * as Report from '../../libs/actions/Report';
+
+import AttachmentView from '../AttachmentView';
+import addEncryptedAuthTokenToURL from '../../libs/addEncryptedAuthTokenToURL';
+import canUseTouchScreen from '../../libs/canUseTouchscreen';
+import CONST from '../../CONST';
+import ONYXKEYS from '../../ONYXKEYS';
+import reportPropTypes from '../../pages/reportPropTypes';
+import reportActionPropTypes from '../../pages/home/report/reportActionPropTypes';
+
+const propTypes = {
+    /** sourceUrl is used to determine the starting index in the array of attachments */
+    sourceURL: PropTypes.string,
+
+    /** Callback to update the parent modal's state with a sourceUrl and name from the attachments array */
+    onNavigate: PropTypes.func,
+
+    /** The report currently being looked at */
+    report: reportPropTypes,
+
+    /** Object of report actions for this report */
+    reportActions: PropTypes.objectOf(PropTypes.shape(reportActionPropTypes)),
+
+};
+
+const defaultProps = {
+    sourceURL: '',
+    report: {
+        isLoadingMoreReportActions: false,
+        reportID: '',
+    },
+    reportActions: {},
+    onNavigate: () => {},
+};
+
+class AttachmentCarousel extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.canUseTouchScreen = canUseTouchScreen();
+        this.makeStateWithReports = this.makeStateWithReports.bind(this);
+        this.cycleThroughAttachments = this.cycleThroughAttachments.bind(this);
+        this.onShowArrow = this.onShowArrow.bind(this);
+
+        this.state = {shouldShowArrow: this.canUseTouchScreen};
+    }
+
+    componentDidMount() {
+        this.makeStateWithReports();
+    }
+
+    componentDidUpdate(prevProps) {
+        const prevReportSize = _.size(prevProps.reportActions);
+        const currReportSize = _.size(this.props.reportActions);
+        if (prevReportSize === currReportSize) {
+            return;
+        }
+        this.makeStateWithReports();
+    }
+
+    /**
+     * Toggles the visibility of the arrows
+     * @param {Boolean} shouldShowArrow
+     */
+    onShowArrow(shouldShowArrow) {
+        this.setState({shouldShowArrow});
+    }
+
+    /**
+     * Helps to navigate between next/previous attachments
+     * @param {Object} attachmentItem
+     * @returns {Object}
+     */
+    getAttachment(attachmentItem) {
+        const sourceURL = _.get(attachmentItem, 'sourceURL', '');
+        const file = _.get(attachmentItem, 'file', {name: ''});
+        return {
+            sourceURL,
+            file,
+        };
+    }
+
+    /**
+     * This is called when there are new reports to set the state
+     */
+    makeStateWithReports() {
```

Comment:
Okay so i think there is an issue with the sourceURL here. Somehow it looks like the domain is removed from the image source. So for example take the image URL as,
> https://www.expensify.com.dev/chat-attachments/6009/w_f7a389e2bfd8df3410a6ba58a9e7ce665a513e9f.jpg?encryptedAuthToken=1o1bQuuvT7MMsQYZOmUWARB0QejsxKLTJBtiqYvTtLPS3kgAS9Pqd%2BwMWJ57T4A6zRfYoF%2FnbzM3A5umpAD9u9mvXQW3eywg4RORihNGaIX14jRSXAeBcSPPk2pVW58EYU6fB4vFcnhFNhrbVnU0gdf8GxU21n9Az4UXfT8ETdtnJ7iYM1gPJ0H16OTRu4k5P1KPbzciPeyzGwQRr8lXZIfjdNuVumWejWJGPzhGaLkd2BsKvact%2F2%2BOHst8jP6rHUzyF9uJg%2BarF%2BZ6jwHMjEeKU80bRyoFDfxc7evHQE61AMmRWzYK3UNM8HNyyVQ%2FWjppbvYk1YTLnq%2FvRibO2%2BrvDLx7K5rjAa%2BdCSSL5%2BoB%2FJeYIb%2F%2Bo8rEzyMKSZ87p69IwhLZK1BxslPfLFmNZl19YAT4csoN0sYqungVpltB0qy2MGjktMes4b0NLXnlsidfjOBUgrsN3%2B1sSsMXRCiqpFf0qyOwt0igeX7wMV4zEj3CRS5ejRAtKvXUotZ9icBdsU5SRj7ngdHi%2BjQmsnPkmwMLZKUyVXF87ERCrV1ojiCxtfXjHwpfXsUFHCPcolCIkkOen7gpjwLvBPNUrDQD4CM%2BD8hbAscDr2o02Ul%2FpJ2iFe%2B5I6Scegqv%2FSu3umB585L8cV%2BxE4WDm8aKN8vLK3Jqqf4jXxvdZ8RFQYQd34mn1Qj5DUc31JA6YP0TjpRwQQgeY6T2UOxRD6NLqAOMX11Fm1JP7pcuzqQrO2SYh4SJ4LCl0uT4w9kf6KiwDhu2aodsh9AqSEmuc2yCAdhR6JnMJCJX757CiqD518IPYLYswVbmz%2BeTwzALgkSCAWdaLgBifXloqMfEkZ6tOBn0rzleDOVeQYR7SRP4s1AlhHGbh%2FKpdbi7CaNjYUcaQFNhZczlWfuq5q%2BOWTZJc76GGzWDfkgSA4%2FjkoXLRZo%3D'

The domain is `www.expensify.com.dev` and its important to retain that. This is because sometimes the image is on production expensify.com, but the user is testing on staging.com. Or for our engineers we test on localhost:8080 but the image is in expensify.com.dev. So if a domain is not provided it assumes localhost:8080 and then fails.

So can you check the logic here to ensure the domain is not lost.